### ISSUE TITLE
Rework return statement

### ISSUE THREAD

**Issue by nickschiffer** [nickschiffer/Pascal-Compiler#21]
ie We don't want to be able to return a function declaration.

**Comment by nickschiffer** [comment]
return expr will work I believe

**Comment by chadwpalm** [comment]
I agree with it needing to be 'return' expr. I also am thinking that return_stmt should be added to the statements at the top of the file. Even though return should only be used inside of a function, it's the only way to have it inside of a compound statement. The parsing logic would deal with where it is allowed to be during parsing, not scanning......but maybe I'm off on all of this.
